src/v/cloud_roles/apply_credentials.cc

```diff
@@ -10,10 +10,10 @@
 
 #include "cloud_roles/apply_credentials.h"
 
-#include "cloud_roles/apply_abs_credentials.h"
-#include "cloud_roles/apply_abs_oauth_credentials.h"
-#include "cloud_roles/apply_aws_credentials.h"
-#include "cloud_roles/apply_gcp_credentials.h"
+#include "apply_abs_credentials.h"
+#include "apply_abs_oauth_credentials.h"
+#include "apply_aws_credentials.h"
+#include "apply_gcp_credentials.h"
```

Comment:
Can you remind me: is the removal of the prefixing module a requirement of becoming a new-style module? Or is this to make it clearer what is internal or external? It'd be great for reviewers and future readers to leave some notes in the commit message about the distinction

Reply:
the public headers are moved into the `<module>/include/<module>` directory. everything else is remaining is now private. that means we have to remove the prefix, otherwise it'd be referring to the headers in the public directory, but these are private.